src/core/raw_tracepoint/v0_6_0.rs

```diff
@@ -0,0 +1,45 @@
+use bcc_sys::bccapi::*;
+use failure::*;
+
+use std::ffi::CString;
+use std::fs::File;
+use std::hash::{Hash, Hasher};
+use std::os::unix::prelude::*;
+
+#[derive(Debug)]
+pub struct RawTracepoint {
+    name: CString,
+    code_fd: File,
+    p: i32,
+}
+
+impl RawTracepoint {
+    pub fn attach_raw_tracepoint(name: &str, file: File) -> Result<Self, Error> {
```

Comment:
I don't know if it's possible, but if you put the conditional compilation here you don't need to have `v0_4_0.rs`, this file can just be named `src/core/raw_tracepoint.rs` and avoid the duplicate implementation of `PartialEq`, `Eq`, and `Hash`. I see that the implementations are different, i just think it's easier to reason about feature-specific logic if it's all in one place for a given code path ie:

```

#![allow(unused_variables)]
fn main() {
let machine_kind = if cfg!(unix) {
  "unix"
} else if cfg!(windows) {
  "windows"
} else {
  "unknown"
};

println!("I'm running on a {} machine!", machine_kind);
}
```

Reply:
Looking back at this comment, I realize that example has noting to do with any of this code. I actually copied it from here: https://doc.rust-lang.org/reference/conditional-compilation.html#the-cfg-macro